hypershift-operator/controllers/machineconfigserver/machineconfigserver_controller.go

```diff
@@ -28,7 +31,11 @@ import (
 )
 
 const (
-	finalizer = "hypershift.openshift.io/finalizer"
+	finalizer                                = "hypershift.openshift.io/finalizer"
+	haproxyConfigSecretName                  = "machine-config-server-haproxy-config"
+	hostedClusterConfigOperatorConfigMapName = "hosted-cluster-config-operator"
+	defaultRouterCAConfigmapNamespace        = "openshift-config-managed"
```

Comment:
assumption: this config map for when the MCS is exposed with a route holds the CA that can be used to validate the router connection if it is not signed by a well known certificate authority (otherwise that will be handled with Root RHCOS certs in OS).  We depend on this secret existing with one cert in it though for the route exposure case which looks to be a valid assumption to make